### PR TITLE
Resize window if passed (bug 1124234)

### PR DESCRIPTION
**Pull request by muffinresearch**
Re-center the window if it's passed in. To see what this feels like in reality see http://jsbin.com/jilido/3/

It's not that smooth but probably better than having an over-size window?

### CODE REVIEW

File: lib/fxpay/settings.js [hunk 1]
```diff
@@ -49,6 +49,8 @@
     },
     // Reference window so tests can swap it out with a stub.
     window: window,
+    winWidth: 276,
```
Comment (kumar303) [inline lib/fxpay/settings.js]: maybe preface this with a comment saying how this is used when there's no native window opener

File: lib/fxpay/utils.js [hunk 2]
```diff
@@ -39,27 +39,37 @@
...
+  exports.getCenteredCoordinates = function(w, h) {
+    // Centering calcs that work on multiple monitors (bug 1122683).
+    var x = window.screenX +
+      Math.max(0, Math.floor((window.innerWidth - w) / 2));
+    var y = window.screenY +
+      Math.max(0, Math.floor((window.innerHeight - h) / 2));
+    return [x, y];
+  };
+
+  exports.reCenterWindow = function(winRef, w, h) {
```
Comment (kumar303) [inline lib/fxpay/utils.js]: it seems weird to me that this arg takes width and height but all other functions in this module use ``settings.winWidth`` directly. One thing you could do is allow optional width/height but fall back to the settings values.

**Comment by kumar303** [comment]
oh man we're deep in it now! No turning back. POPUP MADNESS.

Does this code need to do window focusing?

r+wc

File: lib/fxpay/utils.js (hunk 2 above)
Comment (muffinresearch) [inline lib/fxpay/utils.js]: Fixed.

**Comment by muffinresearch** [comment]
Going to need to do the resizing after we're on the same domain otherwise we don't have access. This is only a problem for new sign-ups.

**Pull request merged**